.github/workflows/workflow.yml

```diff
@@ -87,14 +87,13 @@ jobs:
 
       - name: Define env (unix)
         if: ${{ runner.os != 'Windows' }}
-        run: echo "::set-env name=GITHUB_TAG::${GITHUB_REF#refs/*/}"
+        run: echo "GITHUB_TAG=$(git describe --always)" >> $GITHUB_ENV
 
       - name: Define env (windows)
         if: ${{ runner.os == 'Windows' }}
```

Comment:
If the same method can work for all OSes then we might as well should have a single `Define env` section, and can get rid of this one specific to Windows.

Reply:
Makes sense, I don't have a windows box to test, but the CI should help identify this. If it looks like it can be simplified, I'll take a pass at it.

Reply:
It looks like the windows build isn't picking up the correct `CARMELC_STDLIB_PATH` for Windows. Either with the variable being set in the workflow.yml or just leaving it with the make file. Not 100% sure the right fix yet

Reply:
Alas, I don't have a Windows machine either to test things out :( -- and I haven't been able to cache the Windows build either, which makes this a particularly painful iteration process.

Thanks a lot for working on it 🙌🏼

Reply:
I think I just figured it out. Maybe I can look into caching on the windows side too. One other thought I had was about updating the Makefile such that it shells out to `pwd` which should allow that to work without the custom env var. If you are interested I can squash some of the commits in this PR (with a couple of small clean ups) and spin-up a couple more with different experiments of the windows build.

Reply:
That sounds fantastic, yes :) give to a shot 👍🏼

And feel free to commit as much as you need to, we'll squash and make the commit message pretty at merge time 🦾